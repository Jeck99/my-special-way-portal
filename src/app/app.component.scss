.msw-wrapper {
    direction: rtl;
  
    .main-panel {
      float: left;
      width: 100%;
    }
    .sidebar {
      left: auto;
      right: 0;
    }
  }
  
  .msw-div-column-left{
    margin-right:18px;
  }
  
  .msw-div-column-right{
    margin-left:18px;
  }

  .msw-button-delete.mat-button{
    width: 109.7px;
    height: 36px;
    border-radius: 3px;
    background-color: #ff213c;        

    /deep/ .mat-button-wrapper{
        width: 93.7px;
        height: 16px;
        font-family: Rubik;
        font-size: 14px;
        font-weight: 500;
        font-style: normal;
        font-stretch: normal;
        line-height: 1.14;
        letter-spacing: 1.3px;
        text-align: center;
        color: #ffffff;
    }
}

.msw-button-cancel.mat-button{
    width: 93.7px;
    height: 16px;
    font-family: Rubik;
    font-size: 14.1px;
    font-weight: 500;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.13;
    letter-spacing: 1.3px;
    text-align: center;
    color: #4a4a4a;
}
