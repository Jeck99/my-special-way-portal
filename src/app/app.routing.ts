import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BrowserModule } from '@angular/platform-browser';
import { Routes, RouterModule } from '@angular/router';
import { DashboardComponent } from './pages/dashboard/dashboard.component';
import { AuthGuard } from './services/authentication/auth.guard';
import { LoginComponent } from './pages/login/login.component';
import { StudentComponent } from './pages/student/student.component';
import { ClassComponent } from './pages/class/class.component';
import { UserComponent } from './pages/user/user.component';
import {StudentDetailsComponent} from './pages/student/details/student-details.component';

const routes: Routes = [
  { path: 'dashboard', component: DashboardComponent, canActivate: [AuthGuard] },
  { path: 'class', component: ClassComponent, canActivate: [AuthGuard] },
  { path: 'student', component: StudentComponent, canActivate: [AuthGuard] },
  { path: 'user', component: UserComponent, canActivate: [AuthGuard] },
  { path: 'login', component: LoginComponent },
  { path: '', redirectTo: 'student', pathMatch: 'full' },
  { path: 'student/new', component: StudentDetailsComponent},
  { path: 'student/:id', component: StudentDetailsComponent},
];

@NgModule({
  imports: [
    CommonModule,
    BrowserModule,
    RouterModule.forRoot(routes),
  ],
  exports: [
  ],
})
export class AppRoutingModule { }
