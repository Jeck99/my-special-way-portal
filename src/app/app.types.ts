
export type Message = {
    message: string;
  };
  export type Query = {
      allMessages: Message[];
  };
