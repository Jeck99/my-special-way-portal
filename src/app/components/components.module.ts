import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';

import { FooterComponent } from './footer/footer.component';
import { NavbarComponent } from './navbar/navbar.component';
import { SidebarComponent } from './sidebar/sidebar.component';

import { ApolloModule, Apollo } from 'apollo-angular';
import { HttpLinkModule, HttpLink } from 'apollo-angular-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';

@NgModule({
  imports: [
    CommonModule,
    RouterModule,
    ApolloModule,
    HttpLinkModule
  ],
  declarations: [
    FooterComponent,
    NavbarComponent,
    SidebarComponent
  ],
  exports: [
    FooterComponent,
    NavbarComponent,
    SidebarComponent
  ]
})
export class ComponentsModule {
  constructor(
    apollo: Apollo,
    httpLink: HttpLink
  ) {
    apollo.create({
      link: httpLink.create({ uri: 'https://msw-server.azurewebsites.net/graphql' }),
      cache: new InMemoryCache()
    });
  }
}
