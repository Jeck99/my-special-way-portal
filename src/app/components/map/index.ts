export * from './map.component';
export * from './paths-mapper.pipe';
