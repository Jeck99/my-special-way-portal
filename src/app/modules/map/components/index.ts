export * from './map.component';
export * from './pipes/paths-mapper.pipe';
export * from './services/map.service';
