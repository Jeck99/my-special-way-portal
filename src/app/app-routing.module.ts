import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { MapComponent } from './components';

const routes: Routes = [
  {
    path: 'map',
    component: MapComponent
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
