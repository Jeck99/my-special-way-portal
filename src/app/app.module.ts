import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { LeafletModule } from '@asymmetrik/ngx-leaflet';
import { LeafletDrawModule } from '@asymmetrik/ngx-leaflet-draw';

import { AppComponent } from './app.component';
import {
  MapComponent,
  PathsMapperPipe,
} from './components';

@NgModule({
  declarations: [
    AppComponent,
    MapComponent,
    PathsMapperPipe,
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    LeafletModule.forRoot(), // later move this into map module
    LeafletDrawModule.forRoot(), // later move this into map module
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
