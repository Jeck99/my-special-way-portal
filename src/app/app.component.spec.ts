import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import {CdkTableModule} from '@angular/cdk/table';
import { ApolloModule, Apollo } from 'apollo-angular';
import { HttpLinkModule, HttpLink } from 'apollo-angular-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { ComponentsModule } from './components/components.module';
import {APP_BASE_HREF} from '@angular/common';
import {
  MatButtonModule, MatCardModule, MatDialogModule, MatIconModule,
  MatInputModule, MatPaginatorModule, MatProgressSpinnerModule,
  MatRippleModule, MatSortModule,
  MatTableModule,
  MatTooltipModule
} from '@angular/material';

import { AppRoutingModule } from './app.routing';
import { DashboardComponent } from './pages/dashboard/dashboard.component';
import { AgmCoreModule } from '@agm/core';
import {AlertComponent} from './components/alert/alert.component';
import {AlertService} from './components/alert/services/alert.service';
import {LoginComponent} from './pages/login/login.component';
import {AuthGuard} from './services/auth.guard';
import {AuthenticationService} from './services/authentication.service';
import {HTTP_INTERCEPTORS, HttpClientModule} from '@angular/common/http';
import {JwtInterceptor} from './services/helpers/jwt.interceptor';
import {UserService} from './services/user.service';
import {fakeBackendProvider} from './services/helpers/fake-backend';
import {ClassComponent} from './pages/class/class.component';
import {ClassService} from './pages/class/services/class.service';
import {StudentComponent} from './pages/student/student.component';
import {StudentService} from './pages/student/services/studnet.service';
import {DeleteDialogComponent} from './pages/class/dialogs/delete/delete.dialog.component';


describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        BrowserAnimationsModule,
        FormsModule,
        ComponentsModule,
        RouterTestingModule,
        MatButtonModule,
        MatRippleModule,
        MatInputModule,
        MatTooltipModule,
        MatTableModule,
        MatSortModule,
        MatPaginatorModule,
        MatPaginatorModule,
        MatIconModule,
        MatCardModule,
        MatProgressSpinnerModule,
        MatDialogModule,
        AppRoutingModule,
        CdkTableModule,
        HttpClientModule,
        AgmCoreModule.forRoot({
          apiKey: 'YOUR_GOOGLE_MAPS_API_KEY'
        }),
        ApolloModule,
        HttpLinkModule
      ],
      declarations: [
        AppComponent,
        DashboardComponent,
        AlertComponent,
        LoginComponent,
        ClassComponent,
        StudentComponent,
        DeleteDialogComponent
      ],
      providers: [
        AlertService,
        AuthGuard,
        AuthenticationService,
        UserService,
        {provide: APP_BASE_HREF, useValue: '/'},
        {
          provide: HTTP_INTERCEPTORS,
          useClass: JwtInterceptor,
          multi: true
        },
        ClassService,
        StudentService,
        // provider used to create fake backend
        fakeBackendProvider
      ],
    }).compileComponents();
  }));
  it('should create the app', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  }));
  it('should render "made with love" in footer', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('.footer').textContent).toContain('Welcome to My-Special-W@@y!');
  }));
  it('should render "made with love" in footer', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('.copyright').textContent).toContain('made with love');
  }));
});
