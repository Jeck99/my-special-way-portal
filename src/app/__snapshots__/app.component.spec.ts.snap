// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard component should render component as described in snapshot 1`] = `
<app-root>
  <div
    class="msw-wrapper wrapper"
  >
    
    
    <div
      class="sidebar"
      data-background-color="white"
      data-color="purple"
      data-image="../assets/img/sidebar-1.jpg"
    >
      
      
      <app-sidebar />
      
    
    </div>
    
    
    <div
      class="main-panel"
    >
      
      
      <app-navbar />
      
      
      <router-outlet />
      
      
      <app-footer />
      
    
    </div>
    
  
  </div>
  
</app-root>
`;
